### ISSUE TITLE
Support Databind Event Operations

### ISSUE THREAD

**Issue by poovarasanvasudevan** [flipkart-incubator/proteus#72]
It is good to have such a support like
`
{
        "visible" : "$data == null ? visible : gone"
}
`

**Comment by adityasharat** [comment]
@poovarasanvasudevan thanks for the suggestion. 😃 
This is one of the major items for the next major version of proteus. I will add a roadmap section in the readme. You can keep a watch on #67

**Comment by adityasharat** [comment]
As a workaround until then, you can override any attribute processor or create a new attribute processor to achieve something like this.

**override**

```java
LayoutHandler handler = layoutBuilder.getHandler("view");  // parent of all view types, if you want it for all view types.
handler.addHandler(new Attributes.Attribute("visible"), new AttributeProcessor<View> () {
  public void handle(String key, JsonElement value, View view) {
    if (checkWhatYouWant(value)) {
      doWhatYouWantWithThe(view);
    }
  }
})
```

Similarly, you can add a new attribute processors too. Will also add this to the [FAQ](https://github.com/flipkart-incubator/proteus/wiki/Frequently-asked-questions) section.

**Comment by adityasharat** [comment]
resolved in [v5.0.0 RC 6](https://github.com/flipkart-incubator/proteus/releases/tag/5.0.0-rc6) using formatters and functions.

example:

```
{
  ""visibility": "~@{any.path}${IF_THEN_ELSE(@{a.b.c},'visibile', 'gone')}"
}
```

breaking it down (ignoring `any.path`):
`IF_THEN_ELSE`  ( `@{a.b.c}` , `'visibile'` , `'gone'` )

if **`a.b.c`** is a truthy value (non null) then set visibility to **'visible'** else **'gone'**.
For more functions and formatters check out [this](https://github.com/flipkart-incubator/proteus/blob/layout-class/proteus-core/src/main/java/com/flipkart/android/proteus/Formatter.java)

closing this issue.
